File: scripts/test_solutions/494_target_sum.cpp
class Solution {
public:
    int findTargetSumWays(vector<int>& nums, int target) {
        int total = 0;
        for (int num : nums) total += num;

        if ((total + target) % 2 != 0 || total + target < 0) {
            return 0;
        }

        int subsetSum = (total + target) / 2;
        vector<int> dp(subsetSum + 1, 0);
        dp[0] = 1;

        for (int num : nums) {
            for (int j = subsetSum; j >= num; j--) {
                dp[j] += dp[j - num];
            }
        }
        return dp[subsetSum];
    }
};

File: scripts/test_solutions/494_target_sum.js
var findTargetSumWays = function(nums, target) {
    const sum = nums.reduce((a, b) => a + b, 0);
    if ((sum + target) % 2 !== 0 || sum + target < 0) return 0;
    const subsetSum = (sum + target) / 2;
    const dp = Array(subsetSum + 1).fill(0);
    dp[0] = 1;
    for (const num of nums) {
        for (let j = subsetSum; j >= num; j--) dp[j] += dp[j - num];
    }
    return dp[subsetSum];
};
